### DISCUSSION TITLE
Gatsby local development /api returns 404 preview page

### DISCUSSION THREAD

**Discussion by xmflsct** [vercel/vercel#3977] [General] [answered]
Trying to migrate below:

- Static site from Github Pages to Now
- Cloud functions from Google Cloud to Now Serverless Functions

My full Gatsby source are at root folder `/`, and I created `/api/test/test.js` for my function.
Then I run `now dev` from my `/` folder. Gatsby site development mode is generated without a problem. But when I navigate to `http://localhost:port/api/test/test`, I get Gatsby's 404 preview page. How can I test my function locally together with Gatsby? Thanks.

**Comment by xmflsct** [comment, marked as answer]
After migration from GCP to Now, I did not rewrite much of the script. After changing them to [helpers](https://zeit.co/docs/runtimes#official-runtimes/node-js/node-js-request-and-response-objects), it works now.

**Reply by styfle** [reply]
You can test locally with `now dev`. https://zeit.co/docs/v2/serverless-functions/introduction#local-development

Glad you got it figured out 👍
